cpp/ql/test/library-tests/dataflow/DefaultTaintTracking/annotate_path_to_sink/tainted.ql

```diff
@@ -0,0 +1,88 @@
+/**
+ * This test provides the possibility to annotate elements when they are on a path of a taint flow to a sink.
```

Comment:
I think this is a good idea! I guess this works because `DefaultTaintTracking` has a very generous definition of sinks. It might be good to have a separate tag for nodes that are part of the path, but is not the actual sink?

Reply:
I don't understand your comment.
Indeed, this feature defines and filters on a custom notion of sink (namely `sink` invocations).
Can you try to explain differently what you mean?
I don't feel for the current tests that we need different tags for `on-the-path-but-not-the-sink` and `the-sink`.

However, an annotation `tainted-but-not-on-a-path-to-a-sink` is doable - I just doubt it's usefulness, but am open to be convinced otherwise.

Reply:
Yeah on second thought I don't think we gain anything by using separate tags. This LGTM!